Source/DafnyCore/Dafny.atg

```diff
@@ -1169,7 +1169,11 @@ ModuleExport<ModuleDefinition parent, out ModuleDecl submodule>
     bool isRefining = false;
     ExportSignature exsig;
   .)
-  [ ExportId<out exportId> ]
+  [ IF(IsIdentifier(la.kind) || la.kind==_digits) ExportId<out exportId>   // If the next token is 'least' or 'greatest' use it as the export id, 
+                                                                           // not the beginning of a subsequent extreme predicate declaration
+    (. if (startToken.line != exportId.line) errors.Warning(exportId, "the id for the export declaration is on a different line than the 'export' keyword -- this looks suspicious or confusing");
```

Comment:
I'm puzzled about that. I've seen people writing

```
  export
    name
    provides X
    reveals Y
```
which seems adequate.

I think the error message could be more appropriate - but I don't have a strong opinion, your warning is already great - if the token before the _start_ of a predicate on the same line was "least" or "greatest" and this predicate itself is not "least" or "greatest".
That way, export names could be on a different line.

Reply:
I suggest that if the input includes `export least predicate`, then we should given an error if `least` and `predicate` are on the same line (not if `export` and `least` are on different lines).

Reply:
OK. Now there is a warning if the token is least or greatest, it is in a export declaration, the next token is 'predicate', and 'predicate'  is on the same line as 'least or 'greatest'.

Or, Rustan, did you really mean 'error' instead of 'warn ing' in your comment?

Reply:
Sorry, I meant warning. Good fix.